src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -368,11 +368,7 @@ ResetVisitedFields(HTAB *adjacencyList)
 static bool
 AssociateDistributedTransactionWithBackendProc(TransactionNode *transactionNode)
 {
-#ifdef USE_ASSERT_CHECKING
```

Comment:
These code blocks are not identical behavior. PG_USED_FOR_ASSERTS_ONLY won't prevent GetLocalGroupId from being called unnecessarily in release builds